lib/lz4hc.c

```diff
@@ -287,6 +304,41 @@ LZ4HC_InsertAndGetWiderMatch (
         }   }   }   }
     }  /* while ((matchIndex>=lowLimit) && (nbAttempts)) */
 
+    if (dict == usingDictCtx && nbAttempts && ip - base - lowLimit < MAX_DISTANCE) {
+        ptrdiff_t dictIndexDelta = dictCtx->base - dictCtx->end + lowLimit;
+        dictMatchIndex = dictCtx->hashTable[LZ4HC_hashPtr(ip)];
+        matchIndex = dictMatchIndex + (int)dictIndexDelta;
```

Comment:
minor : `dictIndexDelta` is casted here, from `ptrdiff_t` to `int`.
In 64-bit mode, `ptrdiff_t` is a 64-bit value, so it can, in theory, overflow `int`.

Considering the initialization of `dictIndexDelta`, the value shouldn't be that large.
Box it with an `assert()` then.

Reply:
Yeah, should be safe since indexes in both contexts can't grow past 1 GB. I'll add an assert.